src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/LocalDB.Unix.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+namespace System.Data.SqlClient.SNI
+{
+    internal class LocalDB
+    {
+        internal static string GetLocalDBConnectionString(string localDbInstance)
+        {
+            throw new PlatformNotSupportedException(); // LocalDB is not available for Unix and hence it cannot be supported.
```

Comment:
Nit: how about customizing the error message here to say something along the lines of what's in the comment?